jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -402,6 +402,22 @@ private void leaveSession(HttpSession s)
             LOG.debug("Request {} leaving session {}", this, session);
         session.getSessionHandler().complete(session);
     }
+    
+    /**
+     * A response is being committed for a session, 
+     * potentially write the session out before the
+     * client receives the response.
+     */
+    private void commitSession(HttpSession s)
+    {
+        if (s == null)
```

Comment:
Perhaps rather than a null check, do an `instanceof Session` check instead as that combines a null check with protection for the cast below.